readthedocs/projects/tasks/builds.py

```diff
@@ -626,6 +628,22 @@ def get_valid_artifact_types(self):
                         },
                     )
 
+                if artifact_format_files == 1:
+                    # Rename file as "<project_slug>-<version_slug>.<artifact_type>",
+                    # which is the filename that Proxito serves for offline formats
+                    filename = list_dir[0]
+                    _, extension = filename.rsplit(".")
+                    shutil.move(
```

Comment:
If we have access to the container, we should the move operation inside it. Doesn't look like `shutil.move` moves the content of the symlink, but still, better safe than sorry.

>  In case of symlinks, a new symlink pointing to the target of src will be created as the destination and src will be removed.

https://docs.python.org/3/library/shutil.html#shutil.move

Or wrap this operation, checking each path for valid targets in https://github.com/readthedocs/readthedocs.org/blob/main/readthedocs/core/utils/filesystem.py.

Reply:
It's safe to use `shutil.move` here since we are just renaming the file here, we are not moving it to a different directory or similar. Besides, the `BuildMediaStorageMixin.rclone_sync_directory` will check if the path is a suspicious path as well before uploading it to S3. There is no need to perform the same check here again.

Reply:
We are still allowing a user to rename files in the build server outside the user build directory, we shouldn't allow that. The attack surface is very minimal, but we shouldn't allow that. shutil.move doesn't copy the content if the file is a symlink, but that doesn't apply when the upper directory is a symlink.

Reply:
I added this check in e9a0f081f